qiita_pet/handlers/api_proxy/studies.py

```diff
@@ -291,9 +289,16 @@ def study_files_get_req(user_id, study_id, prep_template_id, artifact_type):
         # sorting prefixes by length to avoid collisions like: 100 1002
         # 10003
         prep_prefixes = sorted(prep_prefixes, key=len, reverse=True)
+        uploaded = get_files_from_uploads_folders(study_id)
         # group files by prefix
-        sfiles = {p: [f for _, f in uploaded if f.startswith(p)]
-                  for p in prep_prefixes}
+        sfiles = defaultdict(list)
+        for p in prep_prefixes:
+            to_remove = []
+            for fid, f in uploaded:
+                if f.startswith(p):
+                    sfiles[p].append(f)
+                    to_remove.append((fid, f))
```

Comment:
I think this can be changed to be:

```
uploaded.remove((fid, f))
```

And you can remove the list-comprehension below that's filtering the uploaded elements.

Reply:
Tried that originally, see previous failing push. The issue is that you are changing a looping variable, while you are looping on it.

Reply:
Ah, I see, thanks!